ddtrace/contrib/rq/__init__.py

```diff
@@ -134,12 +135,17 @@ def traced_queue_enqueue_job(rq, pin, func, instance, args, kwargs):
     else:
         resource = func_name
 
-    with pin.tracer.trace(
-        schematize_messaging_operation("rq.queue.enqueue_job", provider="rq", direction=SpanDirection.OUTBOUND),
+    with core.context_with_data(
+        "rq.queue.enqueue_job",
+        span_name=schematize_messaging_operation(
+            "rq.queue.enqueue_job", provider="rq", direction=SpanDirection.OUTBOUND
+        ),
+        pin=pin,
         service=trace_utils.int_service(pin, config.rq),
         resource=resource,
         span_type=SpanTypes.WORKER,
-    ) as span:
+        call_key="queue.enqueue_job",
+    ) as ctx, ctx[ctx["call_key"]] as span:
         span.set_tag_str(COMPONENT, config.rq.integration_name)
```

Comment:
You'll want to replace these `set_tag_str` calls with a `dispatch` call. That's fine to do in a separate pull request if you prefer.

Reply:
@emmettbutler  sorry didn't see these updates. I added the span tags under "tags" in the context. The only time i used the dispatch was for `worker.perform.job`  where the job.status and job.origin tags needed to be added after the job is performed in order to match the snapshot trace.

Is it advised to always add the tags via the handler?

Reply:
@jessicagamio it's not - your new approach of using the `tags` kwarg on `context_with_data` is better.